service/history/workflowTaskHandlerCallbacks.go

```diff
@@ -853,17 +854,17 @@ func failWorkflowTask(
 	startedEventID int64,
 	wtFailedCause *workflowTaskFailedCause,
 	request *workflowservice.RespondWorkflowTaskCompletedRequest,
-) (workflow.MutableState, error) {
+) (workflow.MutableState, int64, error) {
 
 	// clear any updates we have accumulated so far
 	wfContext.Clear()
 
 	// Reload workflow execution so we can apply the workflow task failure event
 	mutableState, err := wfContext.LoadMutableState(ctx)
 	if err != nil {
-		return nil, err
+		return nil, common.EmptyEventID, err
 	}
-
+	nextEventBatchId := mutableState.GetNextEventID()
 	if _, err = mutableState.AddWorkflowTaskFailedEvent(
```

Comment:
nextEventBatchId == the eventID returned from AddWorkflowTaskFailedEvent() ? 
I think it's probably more explicit to return the ID of the returned event.

Reply:
There is transient workflow task logic that I don't want it to depend on. We might want to change this workflow task failed + workflow failed logic in the future.